### PR TITLE
🤖 Use multiple avatars for multiple new likes or subscribers

### LINKED ISSUE #20039: 🤖 Use multiple avatars for multiple new likes or subscribers
We should show multiple avatars when there are multiple likes or subscribers. When we have:
* two(2) likes/subscribers we show show two avatars (similar to author+site avatars on the reader ref: p1706116664843479/1706116589.530589-slack-C06BWNSR02K)
* three(3) or more likes/subscribers we show show the first 3 as returned by the API (design yWt5gg3nWORhu079Qfv3mS-fi-27_2789)

### PR DESCRIPTION
**Pull request by antonis**
Fixes #20039

## Description
This PR adds multiple avatars for multiple post or comment likes and multiple subscribers  

|Light mode|Dark mode|
|---|---|
|Screenshot_20240207_164315|Screenshot_20240207_164327|

|Landcape|RTL|
|---|---|
|Screenshot_20240207_164346|Screenshot_20240207_164414|

-----

## To Test:

1. Open the notification tab in the Jetpack app
2. *Verify* that 2 avatars appear for 2 likes/subscribes
3. *Verify* that 3 avatars appears for more than 2 likes/subscribes 

-----

## Regression Notes

1. Potential unintended areas of impact

    - Notifications tab

4. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

5. What automated tests I added (or what prevented me from doing so)

    - We plan to add tests after refactoring this part of the code
   
-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist:

- [x] WordPress.com sites and self-hosted Jetpack sites.
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by antonis** [comment]
I've tweaked the dual avatar sizes and updated the alignment (ref p1707316051751449/1707227233.708549-slack-C06BWNSR02K) with https://github.com/wordpress-mobile/WordPress-Android/pull/20133/commits/7d29468c987dce4e4059f22c5d772752233ec5e1

File: WordPress/src/main/java/org/wordpress/android/models/Note.java [hunk 1]
```diff
@@ -215,6 +218,23 @@ public String getIconURL() {
...
+                for (int i = 0; i < bodyArray.length(); i++) {
+                    String iconUrl = JSONUtils.queryJSON(bodyArray, "body[" + i + "].media[0].url", "");
+                    if (iconUrl != null && !iconUrl.isEmpty()) {
+                        iconUrls.add(iconUrl);
+                    }
+                }
+                return iconUrls;
+            }
+        }
+        return null;
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/models/Note.java:235]: This ad-hoc parsing seems reasonable for now, given that this whole file is due for refactoring to Room / FluxC (i.e. it likely isn't worth refactoring in piecemeal).

**Review by mkevins — APPROVED** [review]
The code is sensible, given the current constraints. I tested this on a Pixel 3a (physical device) and it is working as described.

Nice work Antonis!

**Pull request merged**